### PR TITLE
bce/dera - Integrate county-level residential data

### PR DESCRIPTION
**Pull request by elainethale**

### CODE REVIEW

File: src/agents/common.jl [hunk 1]
```diff
@@ -16,6 +16,7 @@ mutable struct HEMData
     index_h::Dimension # customer types
     index_j::Dimension # green tariff technologies
     index_z::Dimension # zone index
+    index_sector::Dimension # zone index
```
Comment (elainethale) [inline src/agents/common.jl:19]: We probably also want to have the index_h-to-sector mapping (https://github.com/nrel-hem/HolisticElectricityModelData.jl/blob/main/src/create_run_inputs.jl#L580) here so we can refer to it as needed in the rest of the code.

File: src/agents/customer_group.jl [hunk 2]
```diff
@@ -345,47 +357,47 @@ function CustomerGroup(input_filename::AbstractString, model_data::HEMData; id =
         ParamArray(
             "Constant",
             (model_data.index_h,),
-            [0.0, 0.0, 0.0]; 
+            vcat(fill(0.0, 66)),
```
Comment (elainethale) [inline src/agents/customer_group.jl:360]: This length should not be hard-coded. For zeros, can just use length(model_data.index_h).

File: src/agents/customer_group.jl [hunk 3]
```diff
@@ -345,47 +357,47 @@ function CustomerGroup(input_filename::AbstractString, model_data::HEMData; id =
...
             description = "Constant in green power uptake function (regression parameter)",
         ),
         ParamArray(
             "GreenPowerPrice_coefficient",
             (model_data.index_h,),
-            [0.0, -0.55, -0.55];
+            vcat(fill(0.0, 64), [-0.55, -0.55]),
```
Comment (elainethale) [inline src/agents/customer_group.jl:366]: To not hard-code this, need to use the index_h_to_sector mapping (which hasn't yet been pulled into HEMData, but should be). Ditto for the non-zero values below.

File: src/agents/regulator.jl [hunk 4]
```diff
@@ -1626,135 +1626,213 @@ function solve_agent_problem!(
...
+        if startswith(String(h_ct), "Res_")
+            h_to_sector[h_ct] = :Residential
+        elseif h_ct == :Commercial
+            h_to_sector[h_ct] = :Commercial
+        elseif h_ct == :Industrial
+            h_to_sector[h_ct] = :Industrial
+        else
+            error("Unknown customer type: $h_ct")
+        end
+    end
```
Comment (elainethale) [inline src/agents/regulator.jl:1641]: Don't recreate this. Use the mapping from HEMData.jl

File: src/agents/customer_group.jl [hunk 5]
```diff
@@ -345,47 +361,47 @@ function CustomerGroup(input_filename::AbstractString, model_data::HEMData; id =
...
             (model_data.index_h,),
-            [0.0, 0.0, 0.0]; 
+            fill(0.0, length(model_data.index_h)),
             description = "Constant in green power uptake function (regression parameter)",
         ),
         ParamArray(
             "GreenPowerPrice_coefficient",
             (model_data.index_h,),
-            [0.0, -0.55, -0.55];
+            vcat(fill(0.0, length(model_data.index_h) - 2), [-0.55, -0.55]),
```
Comment (elainethale) [inline src/agents/customer_group.jl:370]: This doesn't work if model commercial or industrial at county-level. Need to use the h_to_sector map. Same below for non-zero values.

File: src/agents/customer_group.jl [hunk 6]
```diff
@@ -345,47 +361,47 @@ function CustomerGroup(input_filename::AbstractString, model_data::HEMData; id =
...
             description = "Sum of PPA and REC prices (regression parameter)",
         ),
         ParamArray(
             "EnergyRate_coefficient",
             (model_data.index_h,),
-            [0.0, 0.0, 0.0]; 
+            vcat(fill(0.0, length(model_data.index_h) - 2), [0.0, 0.0]),
```
Comment (elainethale) [inline src/agents/customer_group.jl:376]: Since all zeros, can just use length(model_data.index_h).

File: src/agents/customer_group.jl [hunk 7]
```diff
@@ -341,51 +358,77 @@ function CustomerGroup(input_filename::AbstractString, model_data::HEMData; id =
...
 
-    green_sub_model = GreenSubModel(
-        ParamArray(
-            "Constant",
-            (model_data.index_h,),
-            [0.0, 0.0, 0.0]; 
-            description = "Constant in green power uptake function (regression parameter)",
+    function generate_coefficients(index_h, h_to_sector, coefficients)
+        return [
+            get(coefficients, h_to_sector[index], 0.0) for index in index_h
```
Comment (elainethale) [inline src/agents/customer_group.jl:363]: nit: Elsewhere in the code we would write `for h in index_h` instead of `for index in index_h`.

**Review by elainethale — COMMENTED** [review]
Looks good to me. Address the nit if you would like and go ahead and merge.

**Review by HaleyRoss — APPROVED** [review]
Reviewed and ready to merge changes.

**Pull request merged**